src/components/CreateListItemForm.js

```diff
@@ -0,0 +1,91 @@
+import { CreateItem } from "../YataClient";
+import React, { useState } from "react";
+import "./createListItem.css";
+
+const CONTENT_HINT = {
+  EMPTY: {
+    style: "help",
+    text: "Type in a new item between 0 and 100 characters",
+  },
+  TOO_LONG: {
+    style: "help is-danger",
+    text: "New item must be less than 100 characters",
+  },
+  VALID: {
+    style: "help is-success",
+    text: "Press enter to create a new item",
+  },
+};
+
+const DEFAULT_FORM_STATE = {
+  newItem: "",
+  hint: CONTENT_HINT.EMPTY,
+  valid: false,
+  error: null,
+};
+
+function CreateListItemForm(props) {
+  const [formState, setFormState] = useState(DEFAULT_FORM_STATE);
+
+  function handleNewItemChanged(event) {
+    const newItem = event.target.value;
+    const isItemValid = newItem.length > 0 && newItem.length <= 100;
```

Comment:
Since we also check `newItem`'s length below (to render different error messages) I wonder if it would make more sense to do something more like:

```
let isItemValid = false;
...
} else {
   itemHint = CONTENT_HINT.VALID
   itemIsValid = true
}
```

that way we aren't duplicating the code that checks if valid and we don't potentially get drift.

Reply:
You could even go further and simply remove `itemIsValid` and have the submit function check the hint but we might want to have multiple "valid" content hints in the future (ie, it _is_ valid but not recommended)